torch/csrc/distributed/rpc/profiler/server_process_global_profiler.cpp

```diff
@@ -0,0 +1,115 @@
+#include <shared_mutex>
+
+#include <torch/csrc/distributed/rpc/profiler/server_process_global_profiler.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+namespace {
+
+class ProcessGlobalProfilerStateStackEntry;
+
+#if defined(__MACH__)
+// Compiler error: 'shared_timed_mutex' is unavailable: introduced in
+// macOS 10.12
+std::mutex
+#else
+std::shared_timed_mutex
+#endif
```

Comment:
I wonder if we should use `#if` in conjunction with `using` to define a mutex type, so that we don't have to do this four times.

Reply:
Good idea. Updating.